support/ci/cargo_update.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -euo pipefail 
+ 
+# shellcheck source=./support/ci/shared.sh 
+source ./support/ci/shared.sh 
+
+branch="ci/cargo-update-$(date +"%Y%m%d")"
+git checkout -b "$branch"
+
+toolchain="$(get_toolchain)"
+
+install_rustup
+install_rust_toolchain "$toolchain"
+
+echo "--- :box: Cargo Update"
+cargo +"$toolchain" update
+echo "--- :box: Cargo Check"
+cargo +"$toolchain" check --quiet --all --tests
+
+git add Cargo.lock
+
+if [ "$BUILDKITE" != "true" ]; then 
+  echo "Not running in Buildkite. Please verify cargo updated correctly before opening a Pull Request" 
+  exit 0
```

Comment:
If we're legitimately envisioning running this update logic anywhere outside of Buildkite, it would be better to extract the update logic into a separate file / function, and add the `git` parts to something that runs only in Buildkite. 

Otherwise, we should probably just bail out immediately if we're not in Buildkite, and also make not being in Buildkite an actual error. Having it exit 0 whether it's in Buildkite or not seems confusing and might make us want to reuse this script in an inappropriate context.